rollouts/controllers/remotesync_controller.go

```diff
@@ -227,23 +251,28 @@ func (r *RemoteSyncReconciler) updateStatus(ctx context.Context, rs *gitopsv1alp
 
 // patchExternalSync patches the external sync in the remote clusters targeted by
 // the clusterRefs based on the latest revision of the template in the RemoteSync.
-func (r *RemoteSyncReconciler) patchExternalSync(ctx context.Context, client dynamic.Interface, name string, rs *gitopsv1alpha1.RemoteSync) error {
+func (r *RemoteSyncReconciler) patchExternalSync(ctx context.Context, client dynamic.Interface, name string, rs *gitopsv1alpha1.RemoteSync,
+	namespace string) error {
 	logger := klog.FromContext(ctx)
 
-	newRootSync, err := BuildObjectsToApply(rs)
+	gvr, gvk, err := getGvrAndGvk(rs.Spec.Type)
 	if err != nil {
 		return err
 	}
-	data, err := json.Marshal(newRootSync)
+	newSync, err := BuildObjectsToApply(rs, gvk, namespace)
 	if err != nil {
-		return fmt.Errorf("failed to encode root sync to JSON: %w", err)
+		return err
+	}
+	data, err := json.Marshal(newSync)
+	if err != nil {
+		return fmt.Errorf("failed to encode %s to JSON: %w", gvk.Kind, err)
 	}
-	_, err = client.Resource(rootSyncGVR).Namespace(rootSyncNamespace).Patch(ctx, name, types.ApplyPatchType, data, metav1.PatchOptions{FieldManager: name})
+	_, err = client.Resource(gvr).Namespace(namespace).Patch(ctx, name, types.ApplyPatchType, data, metav1.PatchOptions{FieldManager: name})
```

Comment:
Note for us (since this came up during implementation of metadata on RSync objects): Since this is a patch call, this explains why we don't overwrite the Remote RSync objects.